GammaGeeks/tura-backend
# [Starts #188568435] Start signup endpoint

**What does this PR do?**

This PR implements the signup functionality for the Barefoot Nomad application, allowing new users to create accounts securely with their email and password. It includes a signup page, secure password handling, and JWT-based backend authentication.

**Description of the task to be completed:**

- Create a signup page with fields for email, password, and other necessary information.
- Enable users to sign up with valid credentials and display a welcome message upon successful registration, redirecting them to the dashboard.
- Show error messages for invalid signup attempts, such as duplicate email or invalid input.
- Securely handle passwords with proper hashing and validation.
- Use JWT tokens to store login state securely after signup.

**How should this be manually tested?**

1. Clone the repository and navigate to the project directory.
2. Start the backend server and ensure dependencies are installed.
3. Go to the signup page, enter valid email and password details, and verify redirection to the dashboard after successful signup.
4. Attempt to sign up with an existing email or invalid information and confirm the error messages are displayed.

**What are the relevant pivotal tracker stories?**

- **Title**: User Signup via Email and Password
- **ID**: [#188568435](https://www.pivotaltracker.com/n/projects/2725084/stories/188568435)
